### PR TITLE
Add examples for browsing / copying / exporting table data.

### PR DESCRIPTION
**Pull request by tseaver**
Uses #1014 as a base.

### CODE REVIEW

**Comment by dhermes** [comment]
Will address once base is merged

**Comment by tseaver** [comment]
@dhermes rebased after merge of #1014.

File: docs/bigquery-usage.rst [hunk 1]
```diff
@@ -227,6 +227,20 @@ Update all writable metadata for a table
...
+Get rows from a table's data:
+
+.. doctest::
+
+   >>> from gcloud import bigquery
+   >>> client = bigquery.Client()
+   >>> dataset = client.dataset('dataset_name')
+   >>> table = dataset.table(name='person_ages')
+   >>> rows, next_page_token = table.data(max_results=100)  # API request
+   >>> rows.csv.headers
```
Comment (dhermes) [inline docs/bigquery-usage.rst]: Is `csv` always present?

File: docs/bigquery-usage.rst [hunk 2]
```diff
@@ -429,3 +443,112 @@ Poll until the job is complete:
    'done'
    >>> job.ended
    datetime.datetime(2015, 7, 23, 9, 30, 21, 334792, tzinfo=<UTC>)
+
+Exporting data (async)
+~~~~~~~~~~~~~~~~~~~~~~
+
+Start a job exporting a table's data asynchronously to a set of CSV files,
+located on GCloud Storage.  First, create the job locally:
```
Comment (dhermes) [inline docs/bigquery-usage.rst]: We never resolved the `GCloud` vs. `gcloud` vs. `Gcloud` discussion before. Remind me not to LGTM until we at least discuss.

File: docs/bigquery-usage.rst [hunk 3]
```diff
@@ -429,3 +443,112 @@ Poll until the job is complete:
...
+
+.. doctest::
+
+   >>> from gcloud import bigquery
+   >>> client = bigquery.Client()
+   >>> table = dataset.table(name='person_ages')
+   >>> job = table.export_to_storage(bucket_name='bucket-name',
+   ...                               object_name='export-prefix*.csv',
```
Comment (dhermes) [inline docs/bigquery-usage.rst]: Didn't you update this?

File: docs/bigquery-usage.rst (hunk 2 above)
Comment (tseaver) [inline docs/bigquery-usage.rst]: For non-code names, we should probably track the branding in the [official docs](https://cloud.google.com/storage/), which appears to be "Google Cloud Storage". 906e120 updates usage to match.

File: docs/bigquery-usage.rst (hunk 3 above)
Comment (tseaver) [inline docs/bigquery-usage.rst]: copy-pasta.  d5b3c23 updates.

File: docs/bigquery-usage.rst (hunk 1 above)
Comment (tseaver) [inline docs/bigquery-usage.rst]: I had figured that it might be a lazy property (maybe just a flyweight).

Comment (dhermes) [inline docs/bigquery-usage.rst]: What's a flyweight? Why is it relevant to `csv`? There are other input types than `csv` too

Comment (tseaver) [inline docs/bigquery-usage.rst]: I mean to refer to the [GoF flyweight pattern](https://en.wikipedia.org/wiki/Flyweight_pattern).  I imagined that `rows.csv` would be a stateless, iterator-like object which would be useful for generating a CSV output stream from the extracted data.

Comment (dhermes) [inline docs/bigquery-usage.rst]: OK. Will there be a way to access headers without the `csv` property?

**Comment by tseaver** [comment]
@dhermes PTAL.

**Comment by dhermes** [comment]
I have a [comment][1] still un-addressed

> OK. Will there be a way to access headers without the `csv` property?

[1]: https://github.com/GoogleCloudPlatform/gcloud-python/pull/1016#discussion_r35663176

**Comment by tseaver** [comment]
The https://cloud.google.com/bigquery/docs/reference/v2/tabledata/list API doesn't show any schema in the result set, which implies that the fields must correspond to the table's schema.  I had imagined that `rows`, when iterated, would return tuples of the values (one per row).

**Comment by dhermes** [comment]
So how does that relate to my question. Do you plan on implementing access in another form?

**Comment by tseaver** [comment]
If we can predict what people want to do, we can add helpers.  Otherwise, we can document a pattern like:

```python
for row in rows:
     for field, value in zip(table.schema, row):
          do_somthing(field, value)
```

**Comment by dhermes** [comment]
Can you add that snippet to these docs? (With a spelling patch to `do_somthing`)

**Comment by tseaver** [comment]
2624822

**Comment by dhermes** [comment]
LGTM

**Pull request merged**